src/index.js

```diff
@@ -15,70 +14,44 @@ const {
     userFromRoomDropper,
     messageToRoomSender,
     locationToRoomSender,
-    roomMessagesFetcher,
 } = require('./services')
-const { roomRepo } = require('./repos')
-const { Location } = require('./entities')
+const actions = require('./actions')
+const middlewares = require('./middlewares')
 
 const app = express()
 const server = http.createServer(app)
 const io = socketio(server)
 
-const publicDirPath = path.join(__dirname, '../public')
+app.use(middlewares)
 
-app.use(express.static(publicDirPath))
+app.get('/rooms', actions.getRooms)
+app.get('/room/:roomId/messages', actions.getRoomMessages)
 
-app.get('/rooms', async (req, res) => {
-    try {
-        const rooms = await roomRepo.all()
-        res.json({ rooms: rooms.map((r) => r) })
-    } catch (error) {
-        console.error(error)
-        res.status(500).json({ error: error.message })
-    }
-})
+io.on('connection', async (socket) => {
+    const { username, roomTitle } = socket.handshake.query
+    console.log(`User ${username} has been connected`)
 
-app.get('/room/:roomId/messages', async (req, res) => {
-    try {
-        const roomId = req.params.roomId
-        const roomMessages = await roomMessagesFetcher.call(roomId)
-        res.json(roomMessages)
-    } catch (error) {
-        console.error(error)
-        res.status(500).json({ error: error.message })
-    }
-})
+    const { error, user, room } = await userToRoomJoiner.call({
+        username,
+        roomTitle,
+    })
+    if (error) throw error
 
-io.on('connection', (socket) => {
-    console.log('New WebSocket connection.')
-    let context = {
-        user: null,
-        room: null,
+    const context = {
+        user,
+        room,
     }
 
-    socket.on('join', async ({ username, room: roomTitle }, callback) => {
-        const { error, user, room } = await userToRoomJoiner.call({
-            username,
-            roomTitle,
-        })
-        if (error) return callback(error)
-
-        context = {
-            user,
-            room,
-        }
-
-        socket.join(room)
-        socket.emit('message', generateWelcome())
-        socket.broadcast.to(room).emit('message', generateJoined(user))
+    socket.join(context.room)
+    socket.emit('message', generateWelcome())
+    socket.to(context.room).emit('message', generateJoined(user))
```

Comment:
Why did you remove `broadcast`?

Reply:
https://socket.io/docs/v4/emit-cheatsheet
it is redundant

Reply:
But it's there. In the latest version. O_o